collated/main/erik0704.md

```diff
@@ -0,0 +1,1207 @@
+        return description;
+    }
+
+    public String getDescription() {
+        return description.get();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Event)) {
+            return false;
+        }
+
+        // state check
+        Event e = (Event) other;
+        return name.equals(e.name) && date.equals(e.date) && description.equals(e.description);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getName())
+                .append(" Date: ")
+                .append(getDate())
+                .append(" Description: ")
+                .append(getDescription());
+        return builder.toString();
+    }
+}
+```
+###### \java\seedu\address\model\event\EventList.java
+``` java
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.fxmisc.easybind.EasyBind;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.model.event.exceptions.DuplicateEventException;
+import seedu.address.model.event.exceptions.EventNotFoundException;
+
+/**
+ * A list of persons that enforces uniqueness between its elements and does not allow nulls.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Event#equals(Object)
+ * // @see CollectionUtil#elementsAreUnique(Collection)
+ */
+public class EventList implements Iterable<Event> {
+
+    private final ObservableList<Event> internalList = FXCollections.observableArrayList();
+    // used by asObservableList()
+    private final ObservableList<Event> mappedList = EasyBind.map(internalList, (event) -> event);
+
+    /**
+     * Returns true if the list contains an equivalent person as the given argument.
+     */
+    public boolean contains(Event toCheck) {
+        requireNonNull(toCheck);
+        return internalList.contains(toCheck);
+    }
+
+    /**
+     * Adds a person to the list.
+     * TO DECIDE: whether to @throws DuplicateEventException
+     * if the person to add is a duplicate of an existing person in the list.
+     */
+    public void add(Event toAdd) throws DuplicateEventException {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateEventException();
+        }
+        internalList.add(new Event(toAdd));
+    }
+
+    /**
+     * Replaces the event {@code target} in the list with {@code editedPerson}.
+     *
+     * @throws DuplicateEventException if the replacement is equivalent to another existing person in the list.
+     * TODO: @throws EventNotFoundException if {@code target} could not be found in the list.
```

Comment:
@erik0704 some parts of your code seems to be work-in-progress (numerous todo's). Tidy up before the final submission.